tests/ToggleGroup.test.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import ToggleButton from '../src/ui/ToggleButton';
+import ToggleGroup from '../src/ui/ToggleGroup';
+
+const renderComponent = () => {
+  return render(
+    <ToggleGroup defaultValue="sedona">
+      <ToggleButton value="sedona">Sedona</ToggleButton>
+      <ToggleButton value="santa_fe">Santa Fe</ToggleButton>
+      <ToggleButton value="las_cruces">Las Cruces</ToggleButton>
+    </ToggleGroup>
+  );
+};
+
+describe('render toggle group component', () => {
+  test('radiogroup should be in the document', () => {
+    const { getByRole } = renderComponent();
+    expect(getByRole('radiogroup')).toBeInTheDocument();
+  });
+
+  test('toggle group should contain all radio buttons', () => {
+    const { queryAllByRole } = renderComponent();
+    expect(queryAllByRole('radio')).toHaveLength(3);
+  });
+
+  test('triggering a click should alter button state', () => {
+    const { getByLabelText } = renderComponent();
+
+    expect(getByLabelText('Sedona')).toBeChecked();
+
+    fireEvent.click(getByLabelText('Santa Fe'));
+
+    expect(getByLabelText('Sedona')).not.toBeChecked();
+    expect(getByLabelText('Santa Fe')).toBeChecked();
+  });
+
+  test('onChange callback function should be called', () => {
+    const onChangeHandler = jest.fn();
+    const { getByLabelText } = render(
+      <ToggleGroup defaultValue="sedona" onChange={onChangeHandler}>
+        <ToggleButton value="sedona">Sedona</ToggleButton>
+        <ToggleButton value="santa_fe">Santa Fe</ToggleButton>
+        <ToggleButton value="las_cruces">Las Cruces</ToggleButton>
+      </ToggleGroup>
+    );
+
+    fireEvent.click(getByLabelText('Santa Fe'));
+
+    expect(onChangeHandler).toHaveBeenCalledTimes(1);
```

Comment:
You can also test that it was called with `santa_fe` as the first argument using `toHaveBeenCalledWith`

Reply:
Updated, thanks!